### ISSUE TITLE
Cannot connect to remote sypder ipython console

### ISSUE THREAD

**Issue by sungengyi** [spyder-ide/spyder#23021]
### What steps will reproduce the problem?

I have updated my packages and spyder as well. I'm connecting to a server (gpu node) after logging in at the login server. I noticed Sypder 6 had implemented the saved remote connection, but since the server I want to use cannot be accessed directly. I wonder if there is a way I can save my remote connection (gpu node). I have configurated my ssh config file. In VScode if I can directly connect myself to the gpu node once I update my ssh config file.

### Traceback
```python-traceback
Traceback (most recent call last):
  File "C:\Users\sunge\anaconda3\Lib\site-packages\spyder\plugins\ipythonconsole\widgets\shell.py", line 1379, in _handle_kernel_died
    super()._handle_kernel_died(since_last_heartbeat)
  File "C:\Users\sunge\anaconda3\Lib\site-packages\qtconsole\frontend_widget.py", line 532, in _handle_kernel_died
    self._kernel_restarted_message(died=True)
  File "C:\Users\sunge\anaconda3\Lib\site-packages\spyder\plugins\ipythonconsole\widgets\shell.py", line 1394, in _kernel_restarted_message
    "\n" + _("Its crash file is located at:") + " "
TypeError: can only concatenate str (not "NoneType") to str
Traceback (most recent call last):
  File "C:\Users\sunge\anaconda3\Lib\site-packages\spyder\plugins\ipythonconsole\widgets\shell.py", line 1379, in _handle_kernel_died
    super()._handle_kernel_died(since_last_heartbeat)
  File "C:\Users\sunge\anaconda3\Lib\site-packages\qtconsole\frontend_widget.py", line 532, in _handle_kernel_died
    self._kernel_restarted_message(died=True)
  File "C:\Users\sunge\anaconda3\Lib\site-packages\spyder\plugins\ipythonconsole\widgets\shell.py", line 1394, in _kernel_restarted_message
    "\n" + _("Its crash file is located at:") + " "
TypeError: can only concatenate str (not "NoneType") to str
Traceback (most recent call last):
  File "C:\Users\sunge\anaconda3\Lib\site-packages\spyder\plugins\ipythonconsole\widgets\shell.py", line 1379, in _handle_kernel_died
    super()._handle_kernel_died(since_last_heartbeat)
  File "C:\Users\sunge\anaconda3\Lib\site-packages\qtconsole\frontend_widget.py", line 532, in _handle_kernel_died
    self._kernel_restarted_message(died=True)
  File "C:\Users\sunge\anaconda3\Lib\site-packages\spyder\plugins\ipythonconsole\widgets\shell.py", line 1394, in _kernel_restarted_message
    "\n" + _("Its crash file is located at:") + " "
TypeError: can only concatenate str (not "NoneType") to str
Traceback (most recent call last):
  File "C:\Users\sunge\anaconda3\Lib\site-packages\spyder\plugins\ipythonconsole\widgets\shell.py", line 1379, in _handle_kernel_died
    super()._handle_kernel_died(since_last_heartbeat)
  File "C:\Users\sunge\anaconda3\Lib\site-packages\qtconsole\frontend_widget.py", line 532, in _handle_kernel_died
    self._kernel_restarted_message(died=True)
  File "C:\Users\sunge\anaconda3\Lib\site-packages\spyder\plugins\ipythonconsole\widgets\shell.py", line 1394, in _kernel_restarted_message
    "\n" + _("Its crash file is located at:") + " "
TypeError: can only concatenate str (not "NoneType") to str
Traceback (most recent call last):
  File "C:\Users\sunge\anaconda3\Lib\site-packages\spyder\plugins\ipythonconsole\widgets\shell.py", line 1379, in _handle_kernel_died
    super()._handle_kernel_died(since_last_heartbeat)
  File "C:\Users\sunge\anaconda3\Lib\site-packages\qtconsole\frontend_widget.py", line 532, in _handle_kernel_died
    self._kernel_restarted_message(died=True)
  File "C:\Users\sunge\anaconda3\Lib\site-packages\spyder\plugins\ipythonconsole\widgets\shell.py", line 1394, in _kernel_restarted_message
    "\n" + _("Its crash file is located at:") + " "
TypeError: can only concatenate str (not "NoneType") to str
Traceback (most recent call last):
  File "C:\Users\sunge\anaconda3\Lib\site-packages\spyder\plugins\ipythonconsole\widgets\shell.py", line 1379, in _handle_kernel_died
    super()._handle_kernel_died(since_last_heartbeat)
  File "C:\Users\sunge\anaconda3\Lib\site-packages\qtconsole\frontend_widget.py", line 532, in _handle_kernel_died
    self._kernel_restarted_message(died=True)
  File "C:\Users\sunge\anaconda3\Lib\site-packages\spyder\plugins\ipythonconsole\widgets\shell.py", line 1394, in _kernel_restarted_message
    "\n" + _("Its crash file is located at:") + " "
TypeError: can only concatenate str (not "NoneType") to str
```

## Versions

* Spyder version: 6.0.1  (conda)
* Python version: 3.12.7 64-bit
* Qt version: 5.15.2
* PyQt5 version: 5.15.10
* Operating System: Windows-11-10.0.22631-SP0

### Dependencies

```
# Mandatory:
aiohttp >=3.9.3                  :  3.10.5 (OK)
asyncssh >=2.14.0,<3.0.0         :  2.17.0 (OK)
atomicwrites >=1.2.0             :  1.4.0 (OK)
chardet >=2.0.0                  :  4.0.0 (OK)
cloudpickle >=0.5.0              :  3.0.0 (OK)
cookiecutter >=1.6.0             :  2.6.0 (OK)
diff_match_patch >=20181111      :  20200713 (OK)
github >=2.3.0                   :  2.4.0 (OK)
importlib_metadata >=4.6.0       :  7.0.1 (OK)
intervaltree >=3.0.2             :  3.1.0 (OK)
IPython >=8.13.0,<9.0.0,!=8.17.1 :  8.27.0 (OK)
jedi >=0.17.2,<0.20.0            :  0.19.1 (OK)
jellyfish >=0.7                  :  1.0.1 (OK)
jsonschema >=3.2.0               :  4.23.0 (OK)
keyring >=17.0.0                 :  24.3.1 (OK)
nbconvert >=4.0                  :  7.16.4 (OK)
numpydoc >=0.6.0                 :  1.7.0 (OK)
parso >=0.7.0,<0.9.0             :  0.8.3 (OK)
pexpect >=4.4.0                  :  4.8.0 (OK)
pickleshare >=0.4                :  0.7.5 (OK)
psutil >=5.3                     :  5.9.0 (OK)
pygments >=2.0                   :  2.15.1 (OK)
pylint >=3.1,<4                  :  3.2.7 (OK)
pylint_venv >=3.0.2              :  3.0.3 (OK)
pyls_spyder >=0.4.0              :  0.4.0 (OK)
pylsp >=1.12.0,<1.13.0           :  1.12.0 (OK)
pylsp_black >=2.0.0,<3.0.0       :  2.0.0 (OK)
pyuca >=1.2                      :  1.2 (OK)
qdarkstyle >=3.2.0,<3.3.0        :  3.2.3 (OK)
qstylizer >=0.2.2                :  0.2.2 (OK)
qtawesome >=1.3.1,<1.4.0         :  1.3.1 (OK)
qtconsole >=5.6.0,<5.7.0         :  5.6.0 (OK)
qtpy >=2.4.0                     :  2.4.1 (OK)
rtree >=0.9.7                    :  1.0.1 (OK)
setuptools >=49.6.0              :  75.1.0 (OK)
sphinx >=0.6.6                   :  7.3.7 (OK)
spyder_kernels >=3.0.0,<3.1.0    :  3.0.0 (OK)
superqt >=0.6.2,<1.0.0           :  0.6.7 (OK)
textdistance >=4.2.0             :  4.2.1 (OK)
three_merge >=0.1.1              :  0.1.1 (OK)
watchdog >=0.10.3                :  4.0.1 (OK)
yarl >=1.9.4                     :  1.11.0 (OK)
zmq >=24.0.0                     :  25.1.2 (OK)

# Optional:
cython >=0.21                    :  None (NOK)
matplotlib >=3.0.0               :  3.9.2 (OK)
numpy >=1.7                      :  1.26.4 (OK)
pandas >=1.1.1                   :  2.2.2 (OK)
scipy >=0.17.0                   :  1.13.1 (OK)
sympy >=0.7.3                    :  1.13.2 (OK)
```

**Comment by sungengyi** [comment]
https://github.com/spyder-ide/spyder/issues/22743  Same issue here

**Comment by ccordoba12** [comment]
Hey @sungengyi, thanks for reporting. Actually, this is a a duplicate of issue #22555, which was fixed in our 6.0.2 version, released a couple of weeks ago.

Since it's not available in Anaconda yet, you need to create a new environment with Conda-forge packages (which are always up to date). For that you need to close Spyder, open the Anaconda Prompt and run there these commands:

    conda create -n spyder-env -c conda-forge spyder matplotlib pandas python=3.11
    conda activate spyder-env
    spyder

If you need to install more packages for your work (e.g. Scikit-learn), you need to install from Conda-forge as well, like this

    conda activate spyder-env
    conda install -c conda-forge scikit-learn
